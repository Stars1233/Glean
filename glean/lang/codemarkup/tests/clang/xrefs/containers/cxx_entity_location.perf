{
  "@generated": null,
  "facts_searched": {
    "cxx1.DeclarationNameSpan.5": 20,
    "cxx1.EnumDefinition.5": 2,
    "cxx1.FunctionDefinition.5": 1,
    "cxx1.NamespaceDefinition.5": 1,
    "cxx1.RecordDefinition.5": 2,
    "cxx1.Trace.5": 1
  },
  "full_scans": [ "cxx1.Trace.5" ]
}