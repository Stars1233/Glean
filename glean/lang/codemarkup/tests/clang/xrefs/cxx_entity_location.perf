{
  "@generated": null,
  "facts_searched": { },
  "full_scans": [ "codemarkup.FileEntityLocations.30" ]
}