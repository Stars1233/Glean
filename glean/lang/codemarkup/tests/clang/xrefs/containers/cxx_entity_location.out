[
  "@generated",
  {
    "key": {
      "entity": {
        "decl": {
          "typeAlias": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "Ptr" },
                  "scope": {
                    "namespace_": { "key": { "name": { "key": "foo" } } }
                  }
                }
              },
              "type": { "key": "int *" },
              "kind": 1,
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 21,
                "columnBegin": 1,
                "lineEnd": 21,
                "columnEnd": 16
              }
            }
          }
        }
      },
      "location": {
        "name": "Ptr",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 21,
            "columnBegin": 1,
            "lineEnd": 21,
            "columnEnd": 16
          }
        },
        "destination": {
          "file": { "key": "test.cpp" },
          "span": { "start": 337, "length": 3 }
        }
      }
    }
  },
  {
    "key": {
      "entity": {
        "decl": {
          "typeAlias": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "this_is_a_typedef" },
                  "scope": {
                    "namespace_": { "key": { "name": { "key": "foo" } } }
                  }
                }
              },
              "type": { "key": "int" },
              "kind": 0,
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 19,
                "columnBegin": 1,
                "lineEnd": 19,
                "columnEnd": 29
              }
            }
          }
        }
      },
      "location": {
        "name": "this_is_a_typedef",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 19,
            "columnBegin": 1,
            "lineEnd": 19,
            "columnEnd": 29
          }
        },
        "destination": {
          "file": { "key": "test.cpp" },
          "span": { "start": 311, "length": 17 }
        }
      }
    }
  },
  {
    "key": {
      "entity": {
        "decl": {
          "variable": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "a" },
                  "scope": {
                    "local": {
                      "key": {
                        "name": {
                          "key": { "name": { "key": "this_is_a_function" } }
                        },
                        "scope": {
                          "namespace_": { "key": { "name": { "key": "foo" } } }
                        }
                      }
                    }
                  }
                }
              },
              "type": { "key": "int" },
              "kind": { "local": { "kind": 2, "attribute": 0 } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 25,
                "columnBegin": 24,
                "lineEnd": 25,
                "columnEnd": 28
              }
            }
          }
        }
      },
      "location": {
        "name": "a",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 25,
            "columnBegin": 24,
            "lineEnd": 25,
            "columnEnd": 28
          }
        },
        "destination": {
          "file": { "key": "test.cpp" },
          "span": { "start": 404, "length": 1 }
        }
      }
    }
  },
  {
    "key": {
      "entity": {
        "decl": {
          "variable": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "b" },
                  "scope": {
                    "local": {
                      "key": {
                        "name": {
                          "key": { "name": { "key": "this_is_a_function" } }
                        },
                        "scope": {
                          "namespace_": { "key": { "name": { "key": "foo" } } }
                        }
                      }
                    }
                  }
                }
              },
              "type": { "key": "int" },
              "kind": { "local": { "kind": 2, "attribute": 0 } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 25,
                "columnBegin": 31,
                "lineEnd": 25,
                "columnEnd": 35
              }
            }
          }
        }
      },
      "location": {
        "name": "b",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 25,
            "columnBegin": 31,
            "lineEnd": 25,
            "columnEnd": 35
          }
        },
        "destination": {
          "file": { "key": "test.cpp" },
          "span": { "start": 411, "length": 1 }
        }
      }
    }
  },
  {
    "key": {
      "entity": {
        "decl": {
          "variable": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "c" },
                  "scope": {
                    "recordWithAccess": {
                      "record": {
                        "key": {
                          "name": { "key": "U" },
                          "scope": {
                            "namespace_": {
                              "key": { "name": { "key": "foo" } }
                            }
                          }
                        }
                      },
                      "access": 0
                    }
                  }
                }
              },
              "type": { "key": "double" },
              "kind": { "field": { "mutable_": false } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 16,
                "columnBegin": 5,
                "lineEnd": 16,
                "columnEnd": 12
              }
            }
          }
        }
      },
      "location": {
        "name": "c",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 16,
            "columnBegin": 5,
            "lineEnd": 16,
            "columnEnd": 12
          }
        },
        "destination": {
          "file": { "key": "test.cpp" },
          "span": { "start": 292, "length": 1 }
        }
      }
    }
  },
  {
    "key": {
      "entity": {
        "decl": {
          "variable": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "n" },
                  "scope": {
                    "recordWithAccess": {
                      "record": {
                        "key": {
                          "name": { "key": "S" },
                          "scope": {
                            "namespace_": {
                              "key": { "name": { "key": "foo" } }
                            }
                          }
                        }
                      },
                      "access": 0
                    }
                  }
                }
              },
              "type": { "key": "int" },
              "kind": { "field": { "mutable_": false } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 11,
                "columnBegin": 12,
                "lineEnd": 11,
                "columnEnd": 16
              }
            }
          }
        }
      },
      "location": {
        "name": "n",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 11,
            "columnBegin": 12,
            "lineEnd": 11,
            "columnEnd": 16
          }
        },
        "destination": {
          "file": { "key": "test.cpp" },
          "span": { "start": 253, "length": 1 }
        }
      }
    }
  },
  {
    "key": {
      "entity": {
        "decl": {
          "variable": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "n" },
                  "scope": {
                    "recordWithAccess": {
                      "record": {
                        "key": {
                          "name": { "key": "U" },
                          "scope": {
                            "namespace_": {
                              "key": { "name": { "key": "foo" } }
                            }
                          }
                        }
                      },
                      "access": 0
                    }
                  }
                }
              },
              "type": { "key": "int" },
              "kind": { "field": { "mutable_": false } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 15,
                "columnBegin": 5,
                "lineEnd": 15,
                "columnEnd": 9
              }
            }
          }
        }
      },
      "location": {
        "name": "n",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 15,
            "columnBegin": 5,
            "lineEnd": 15,
            "columnEnd": 9
          }
        },
        "destination": {
          "file": { "key": "test.cpp" },
          "span": { "start": 278, "length": 1 }
        }
      }
    }
  },
  {
    "key": {
      "entity": {
        "decl": {
          "variable": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "this_is_a_variable" },
                  "scope": {
                    "local": {
                      "key": {
                        "name": {
                          "key": { "name": { "key": "this_is_a_function" } }
                        },
                        "scope": {
                          "namespace_": { "key": { "name": { "key": "foo" } } }
                        }
                      }
                    }
                  }
                }
              },
              "type": { "key": "int" },
              "kind": { "local": { "kind": 0, "attribute": 0 } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 27,
                "columnBegin": 5,
                "lineEnd": 27,
                "columnEnd": 34
              }
            }
          }
        }
      },
      "location": {
        "name": "this_is_a_variable",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 27,
            "columnBegin": 5,
            "lineEnd": 27,
            "columnEnd": 34
          }
        },
        "destination": {
          "file": { "key": "test.cpp" },
          "span": { "start": 424, "length": 18 }
        }
      }
    }
  },
  {
    "key": {
      "entity": {
        "defn": {
          "enum_": {
            "key": {
              "declaration": {
                "key": {
                  "name": {
                    "key": {
                      "name": { "key": "E" },
                      "scope": {
                        "namespace_": { "key": { "name": { "key": "foo" } } }
                      }
                    }
                  },
                  "isScoped": false,
                  "source": {
                    "file": { "key": "test.cpp" },
                    "lineBegin": 23,
                    "columnBegin": 1,
                    "lineEnd": 23,
                    "columnEnd": 24
                  }
                }
              },
              "enumerators": [
                {
                  "key": {
                    "name": { "key": "e_a" },
                    "enumeration": {
                      "key": {
                        "name": {
                          "key": {
                            "name": { "key": "E" },
                            "scope": {
                              "namespace_": {
                                "key": { "name": { "key": "foo" } }
                              }
                            }
                          }
                        },
                        "isScoped": false,
                        "source": {
                          "file": { "key": "test.cpp" },
                          "lineBegin": 23,
                          "columnBegin": 1,
                          "lineEnd": 23,
                          "columnEnd": 24
                        }
                      }
                    },
                    "source": {
                      "file": { "key": "test.cpp" },
                      "lineBegin": 23,
                      "columnBegin": 10,
                      "lineEnd": 23,
                      "columnEnd": 12
                    }
                  }
                },
                {
                  "key": {
                    "name": { "key": "e_b" },
                    "enumeration": {
                      "key": {
                        "name": {
                          "key": {
                            "name": { "key": "E" },
                            "scope": {
                              "namespace_": {
                                "key": { "name": { "key": "foo" } }
                              }
                            }
                          }
                        },
                        "isScoped": false,
                        "source": {
                          "file": { "key": "test.cpp" },
                          "lineBegin": 23,
                          "columnBegin": 1,
                          "lineEnd": 23,
                          "columnEnd": 24
                        }
                      }
                    },
                    "source": {
                      "file": { "key": "test.cpp" },
                      "lineBegin": 23,
                      "columnBegin": 15,
                      "lineEnd": 23,
                      "columnEnd": 17
                    }
                  }
                },
                {
                  "key": {
                    "name": { "key": "e_c" },
                    "enumeration": {
                      "key": {
                        "name": {
                          "key": {
                            "name": { "key": "E" },
                            "scope": {
                              "namespace_": {
                                "key": { "name": { "key": "foo" } }
                              }
                            }
                          }
                        },
                        "isScoped": false,
                        "source": {
                          "file": { "key": "test.cpp" },
                          "lineBegin": 23,
                          "columnBegin": 1,
                          "lineEnd": 23,
                          "columnEnd": 24
                        }
                      }
                    },
                    "source": {
                      "file": { "key": "test.cpp" },
                      "lineBegin": 23,
                      "columnBegin": 20,
                      "lineEnd": 23,
                      "columnEnd": 22
                    }
                  }
                }
              ]
            }
          }
        }
      },
      "location": {
        "name": "E",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 23,
            "columnBegin": 1,
            "lineEnd": 23,
            "columnEnd": 24
          }
        },
        "destination": {
          "file": { "key": "test.cpp" },
          "span": { "start": 355, "length": 1 }
        }
      }
    }
  },
  {
    "key": {
      "entity": {
        "defn": {
          "function_": {
            "key": {
              "declaration": {
                "key": {
                  "name": {
                    "key": {
                      "name": {
                        "key": { "name": { "key": "this_is_a_function" } }
                      },
                      "scope": {
                        "namespace_": { "key": { "name": { "key": "foo" } } }
                      }
                    }
                  },
                  "signature": {
                    "key": {
                      "returns": { "key": "int" },
                      "parameters": [
                        { "name": { "key": "a" }, "type": { "key": "int" } },
                        { "name": { "key": "b" }, "type": { "key": "int" } }
                      ]
                    }
                  },
                  "source": {
                    "file": { "key": "test.cpp" },
                    "lineBegin": 25,
                    "columnBegin": 1,
                    "lineEnd": 29,
                    "columnEnd": 1
                  }
                }
              },
              "isInline": false
            }
          }
        }
      },
      "location": {
        "name": "this_is_a_function",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 25,
            "columnBegin": 1,
            "lineEnd": 29,
            "columnEnd": 1
          }
        },
        "destination": {
          "file": { "key": "test.cpp" },
          "span": { "start": 381, "length": 18 }
        }
      }
    }
  },
  {
    "key": {
      "entity": {
        "defn": {
          "namespace_": {
            "key": {
              "declaration": {
                "key": {
                  "name": { "key": { "name": { "key": "foo" } } },
                  "source": {
                    "file": { "key": "test.cpp" },
                    "lineBegin": 9,
                    "columnBegin": 1,
                    "lineEnd": 31,
                    "columnEnd": 1
                  }
                }
              },
              "members": {
                "key": [
                  {
                    "record_": {
                      "key": {
                        "name": {
                          "key": {
                            "name": { "key": "S" },
                            "scope": {
                              "namespace_": {
                                "key": { "name": { "key": "foo" } }
                              }
                            }
                          }
                        },
                        "kind": { "struct_": { } },
                        "source": {
                          "file": { "key": "test.cpp" },
                          "lineBegin": 11,
                          "columnBegin": 1,
                          "lineEnd": 11,
                          "columnEnd": 19
                        }
                      }
                    }
                  },
                  {
                    "record_": {
                      "key": {
                        "name": {
                          "key": {
                            "name": { "key": "U" },
                            "scope": {
                              "namespace_": {
                                "key": { "name": { "key": "foo" } }
                              }
                            }
                          }
                        },
                        "kind": { "union_": { } },
                        "source": {
                          "file": { "key": "test.cpp" },
                          "lineBegin": 13,
                          "columnBegin": 1,
                          "lineEnd": 17,
                          "columnEnd": 1
                        }
                      }
                    }
                  },
                  {
                    "typeAlias": {
                      "key": {
                        "name": {
                          "key": {
                            "name": { "key": "this_is_a_typedef" },
                            "scope": {
                              "namespace_": {
                                "key": { "name": { "key": "foo" } }
                              }
                            }
                          }
                        },
                        "type": { "key": "int" },
                        "kind": 0,
                        "source": {
                          "file": { "key": "test.cpp" },
                          "lineBegin": 19,
                          "columnBegin": 1,
                          "lineEnd": 19,
                          "columnEnd": 29
                        }
                      }
                    }
                  },
                  {
                    "typeAlias": {
                      "key": {
                        "name": {
                          "key": {
                            "name": { "key": "Ptr" },
                            "scope": {
                              "namespace_": {
                                "key": { "name": { "key": "foo" } }
                              }
                            }
                          }
                        },
                        "type": { "key": "int *" },
                        "kind": 1,
                        "source": {
                          "file": { "key": "test.cpp" },
                          "lineBegin": 21,
                          "columnBegin": 1,
                          "lineEnd": 21,
                          "columnEnd": 16
                        }
                      }
                    }
                  },
                  {
                    "enum_": {
                      "key": {
                        "name": {
                          "key": {
                            "name": { "key": "E" },
                            "scope": {
                              "namespace_": {
                                "key": { "name": { "key": "foo" } }
                              }
                            }
                          }
                        },
                        "isScoped": false,
                        "source": {
                          "file": { "key": "test.cpp" },
                          "lineBegin": 23,
                          "columnBegin": 1,
                          "lineEnd": 23,
                          "columnEnd": 24
                        }
                      }
                    }
                  },
                  {
                    "function_": {
                      "key": {
                        "name": {
                          "key": {
                            "name": {
                              "key": { "name": { "key": "this_is_a_function" } }
                            },
                            "scope": {
                              "namespace_": {
                                "key": { "name": { "key": "foo" } }
                              }
                            }
                          }
                        },
                        "signature": {
                          "key": {
                            "returns": { "key": "int" },
                            "parameters": [
                              {
                                "name": { "key": "a" },
                                "type": { "key": "int" }
                              },
                              {
                                "name": { "key": "b" },
                                "type": { "key": "int" }
                              }
                            ]
                          }
                        },
                        "source": {
                          "file": { "key": "test.cpp" },
                          "lineBegin": 25,
                          "columnBegin": 1,
                          "lineEnd": 29,
                          "columnEnd": 1
                        }
                      }
                    }
                  }
                ]
              }
            }
          }
        }
      },
      "location": {
        "name": "foo",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 9,
            "columnBegin": 1,
            "lineEnd": 31,
            "columnEnd": 1
          }
        },
        "destination": {
          "file": { "key": "test.cpp" },
          "span": { "start": 231, "length": 3 }
        }
      }
    }
  },
  {
    "key": {
      "entity": {
        "defn": {
          "record_": {
            "key": {
              "declaration": {
                "key": {
                  "name": {
                    "key": {
                      "name": { "key": "S" },
                      "scope": {
                        "namespace_": { "key": { "name": { "key": "foo" } } }
                      }
                    }
                  },
                  "kind": { "struct_": { } },
                  "source": {
                    "file": { "key": "test.cpp" },
                    "lineBegin": 11,
                    "columnBegin": 1,
                    "lineEnd": 11,
                    "columnEnd": 19
                  }
                }
              },
              "bases": [ ],
              "members": {
                "key": [
                  {
                    "variable": {
                      "key": {
                        "name": {
                          "key": {
                            "name": { "key": "n" },
                            "scope": {
                              "recordWithAccess": {
                                "record": {
                                  "key": {
                                    "name": { "key": "S" },
                                    "scope": {
                                      "namespace_": {
                                        "key": { "name": { "key": "foo" } }
                                      }
                                    }
                                  }
                                },
                                "access": 0
                              }
                            }
                          }
                        },
                        "type": { "key": "int" },
                        "kind": { "field": { "mutable_": false } },
                        "source": {
                          "file": { "key": "test.cpp" },
                          "lineBegin": 11,
                          "columnBegin": 12,
                          "lineEnd": 11,
                          "columnEnd": 16
                        }
                      }
                    }
                  }
                ]
              }
            }
          }
        }
      },
      "location": {
        "name": "S",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 11,
            "columnBegin": 1,
            "lineEnd": 11,
            "columnEnd": 19
          }
        },
        "destination": {
          "file": { "key": "test.cpp" },
          "span": { "start": 245, "length": 1 }
        }
      }
    }
  },
  {
    "key": {
      "entity": {
        "defn": {
          "record_": {
            "key": {
              "declaration": {
                "key": {
                  "name": {
                    "key": {
                      "name": { "key": "U" },
                      "scope": {
                        "namespace_": { "key": { "name": { "key": "foo" } } }
                      }
                    }
                  },
                  "kind": { "union_": { } },
                  "source": {
                    "file": { "key": "test.cpp" },
                    "lineBegin": 13,
                    "columnBegin": 1,
                    "lineEnd": 17,
                    "columnEnd": 1
                  }
                }
              },
              "bases": [ ],
              "members": {
                "key": [
                  {
                    "variable": {
                      "key": {
                        "name": {
                          "key": {
                            "name": { "key": "n" },
                            "scope": {
                              "recordWithAccess": {
                                "record": {
                                  "key": {
                                    "name": { "key": "U" },
                                    "scope": {
                                      "namespace_": {
                                        "key": { "name": { "key": "foo" } }
                                      }
                                    }
                                  }
                                },
                                "access": 0
                              }
                            }
                          }
                        },
                        "type": { "key": "int" },
                        "kind": { "field": { "mutable_": false } },
                        "source": {
                          "file": { "key": "test.cpp" },
                          "lineBegin": 15,
                          "columnBegin": 5,
                          "lineEnd": 15,
                          "columnEnd": 9
                        }
                      }
                    }
                  },
                  {
                    "variable": {
                      "key": {
                        "name": {
                          "key": {
                            "name": { "key": "c" },
                            "scope": {
                              "recordWithAccess": {
                                "record": {
                                  "key": {
                                    "name": { "key": "U" },
                                    "scope": {
                                      "namespace_": {
                                        "key": { "name": { "key": "foo" } }
                                      }
                                    }
                                  }
                                },
                                "access": 0
                              }
                            }
                          }
                        },
                        "type": { "key": "double" },
                        "kind": { "field": { "mutable_": false } },
                        "source": {
                          "file": { "key": "test.cpp" },
                          "lineBegin": 16,
                          "columnBegin": 5,
                          "lineEnd": 16,
                          "columnEnd": 12
                        }
                      }
                    }
                  }
                ]
              }
            }
          }
        }
      },
      "location": {
        "name": "U",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 13,
            "columnBegin": 1,
            "lineEnd": 17,
            "columnEnd": 1
          }
        },
        "destination": {
          "file": { "key": "test.cpp" },
          "span": { "start": 266, "length": 1 }
        }
      }
    }
  },
  {
    "key": {
      "entity": {
        "enumerator": {
          "key": {
            "name": { "key": "e_a" },
            "enumeration": {
              "key": {
                "name": {
                  "key": {
                    "name": { "key": "E" },
                    "scope": {
                      "namespace_": { "key": { "name": { "key": "foo" } } }
                    }
                  }
                },
                "isScoped": false,
                "source": {
                  "file": { "key": "test.cpp" },
                  "lineBegin": 23,
                  "columnBegin": 1,
                  "lineEnd": 23,
                  "columnEnd": 24
                }
              }
            },
            "source": {
              "file": { "key": "test.cpp" },
              "lineBegin": 23,
              "columnBegin": 10,
              "lineEnd": 23,
              "columnEnd": 12
            }
          }
        }
      },
      "location": {
        "name": "e_a",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 23,
            "columnBegin": 10,
            "lineEnd": 23,
            "columnEnd": 12
          }
        }
      }
    }
  },
  {
    "key": {
      "entity": {
        "enumerator": {
          "key": {
            "name": { "key": "e_b" },
            "enumeration": {
              "key": {
                "name": {
                  "key": {
                    "name": { "key": "E" },
                    "scope": {
                      "namespace_": { "key": { "name": { "key": "foo" } } }
                    }
                  }
                },
                "isScoped": false,
                "source": {
                  "file": { "key": "test.cpp" },
                  "lineBegin": 23,
                  "columnBegin": 1,
                  "lineEnd": 23,
                  "columnEnd": 24
                }
              }
            },
            "source": {
              "file": { "key": "test.cpp" },
              "lineBegin": 23,
              "columnBegin": 15,
              "lineEnd": 23,
              "columnEnd": 17
            }
          }
        }
      },
      "location": {
        "name": "e_b",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 23,
            "columnBegin": 15,
            "lineEnd": 23,
            "columnEnd": 17
          }
        }
      }
    }
  },
  {
    "key": {
      "entity": {
        "enumerator": {
          "key": {
            "name": { "key": "e_c" },
            "enumeration": {
              "key": {
                "name": {
                  "key": {
                    "name": { "key": "E" },
                    "scope": {
                      "namespace_": { "key": { "name": { "key": "foo" } } }
                    }
                  }
                },
                "isScoped": false,
                "source": {
                  "file": { "key": "test.cpp" },
                  "lineBegin": 23,
                  "columnBegin": 1,
                  "lineEnd": 23,
                  "columnEnd": 24
                }
              }
            },
            "source": {
              "file": { "key": "test.cpp" },
              "lineBegin": 23,
              "columnBegin": 20,
              "lineEnd": 23,
              "columnEnd": 22
            }
          }
        }
      },
      "location": {
        "name": "e_c",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 23,
            "columnBegin": 20,
            "lineEnd": 23,
            "columnEnd": 22
          }
        }
      }
    }
  }
]